pkg/apis/gcp/validation/infrastructure.go

```diff
@@ -124,6 +124,10 @@ func ValidateInfrastructureConfig(infra *apisgcp.InfrastructureConfig, nodesCIDR
 		}
 	}
 
+	if infra.Networks.CloudNAT != nil && len(infra.Networks.CloudNAT.NatIPNames) == 0 {
+		allErrs = append(allErrs, field.Forbidden(networksPath.Child("cloudNAT", "natIPNames"), "should not be empty array"))
```

Comment:
Nit: we could use `field.Invalid` instead of `field.Forbidden`:

```suggestion
		allErrs = append(allErrs, field.Invalid(networksPath.Child("cloudNAT", "natIPNames"), infra.Networks.CloudNAT.NatIPNames, "nat IP names cannot be empty"))
```